stable/database/files/nuosm

```diff
@@ -39,6 +39,11 @@ export NUOCMD DB_NAME
 function die() {
   # cleanup
   $NUOCMD set value --key $first_req --value '' --expected-value $HOSTNAME
+  $NUOCMD set value --key $startup_key/$DB_NAME --value '' --expected-value $HOSTNAME
```

Comment:
this comment applies to the file, not to this line.

When I was doing the 24x7 training, I noticed that if the admin is whacked, this script exits with a really cryptic message. The SM itself or nuodocker has more reasonable logging and can be more helpful.

That said, I would like to add some early sanity checks to this file before trying to set values in the KV store. Maybe it is work for another day. 

It is impossible to debug any failures in this script, unless you are REALLY familiar with nuocmd.

Reply:
I have added a test for the admin layer readiness, and exit immediately with an explanatory message that includes whatever error `nuocmd` returned.

Your comments regarding familiarity with `nuocmd` are reasonable, and confirm that `nuosm` relies as much as possible on the existing (relatively low-level) API of `nuocmd`.

I have added some additional diagnostics to the error message.

I am also adding a new file of SM startup tests that verify the correct operation of the script in a number of failure scenarios.